### PR TITLE
feat: show deprecation warning for `https`/`http2` option

### PR DESCRIPTION
**Pull request by snitin315**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
yes

### Motivation / Use-Case

show deprecation warning for `https`/`http2` option.

### Breaking Changes
None

### Additional Info
No

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -523,6 +523,16 @@ class Server {
...
 
+    if (isHTTPs || isSPDY) {
+      util.deprecate(
+        () => {},
+        `'${
+          isHTTPs ? "https" : "http2"
+        }' option is deprecated. Please use the 'server' option.`,
+        `DEP_WEBPACK_DEV_SERVER_${isHTTPs ? "HTTPS" : "HTTP2"}`
+      )();
+    }
```
Comment (alexander-akait) [inline lib/Server.js:534]: Let's add `TODO: remove in the next major release`

**Review by alexander-akait — APPROVED** [review]
We should merge it using `feat:` because it is deprecation

**Review by alexander-akait — COMMENTED** [review]
Can we add example for https/http2 and server options, so we will provide them in our changelog for easy migration?

**Comment by snitin315** [comment]
Added examples in https://github.com/webpack/webpack-dev-server/pull/4007. 

Also, we can link to docs in migration for more information - https://webpack.js.org/configuration/dev-server/#devserverserver

**Pull request merged**